lib/pf/vlan/custom_example.pm

```diff
@@ -1,411 +0,0 @@
-    else {
-        $vlan_number = $switch->getVlanByName('registration');
-    }
-
-    if (!defined($node_info)) {
-        $logger->info("MAC: $mac doesn't have a node entry; belongs into registration VLAN");
-        # CUSTOM: replaced $switch->getVlanByName('registration') with pre-computed VLAN above
-        return $vlan_number;
-    }
-
-    my $n_status = $node_info->{'status'};
-    if ($n_status eq $pf::node::STATUS_UNREGISTERED || $n_status eq $pf::node::STATUS_PENDING) {
-        $logger->info("MAC: $mac is of status $n_status; belongs into registration VLAN");
-        # CUSTOM: replaced $switch->getVlanByName('registration') with pre-computed VLAN above
-        return $vlan_number;
-    }
-    return 0;
-}
-
-=item getNormalVlan
-
-Sample getNormalVlan, see pf::vlan for getNormalVlan interface description
-
-=cut
-
-sub getNormalVlan {
-
-    my ($self, $switch, $ifIndex, $mac, $node_info, $connection_type, $user_name, $ssid, $called_station_id) = @_;
-    my $logger = get_logger();
-
-    # CUSTOM: fetching per building VLAN id if switch type is Cisco::WLC
-    # by default no buliding id vlan prefix
-    my $vlan_id_prefix = '';
-    if (defined($called_station_id) && ref($switch) eq 'pf::Switch::Cisco::WLC_4400') {
-        $vlan_id_prefix = buildingnum_per_called_station_id($called_station_id);
-    }
-
-    # CUSTOM example: admin category
-    # return customVlan to nodes based on category
-    if (defined($node_info->{'category'}) && lc($node_info->{'category'}) eq "student") {
-        return $vlan_id_prefix . $switch->getVlanByName('customVlan2');
-    }
-    elsif (defined($node_info->{'category'}) && lc($node_info->{'category'}) eq "staff")
-    {
-        return $vlan_id_prefix . $switch->getVlanByName('customVlan1');
-    }
-    elsif (defined($node_info->{'category'}) && lc($node_info->{'category'}) eq "guest")
-    {
-        return $vlan_id_prefix . $switch->getVlanByName('customVlan2');
-    }
-
-    $logger->warn("User was not assigned a catagory...Rejecting");
-    return -1;
-
-}
-
-=back
-
-=head1 AUTHOR
-
-Inverse inc. <info@inverse.ca>
-
-=head1 COPYRIGHT
-
-Copyright (C) 2005-2015 Inverse inc.
-
-=head1 LICENSE
-
-This program is free software; you can redistribute it and/or
-modify it under the terms of the GNU General Public License
-as published by the Free Software Foundation; either version 2
-of the License, or (at your option) any later version.
-
-This program is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-GNU General Public License for more details.
-
-You should have received a copy of the GNU General Public License
-along with this program; if not, write to the Free Software
-Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
-USA.
-
-=cut
-
-1;
-
-# vim: set shiftwidth=4:
-# vim: set expandtab:
-# vim: set backspace=indent,eol,start:
```

Comment:
Since we can do it with vlan filters i think it's useless to have old stuff that can disturb